Detectors/TRD/base/src/TrackletTransformer.cxx

```diff
@@ -0,0 +1,121 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "TRDBase/TrackletTransformer.h"
+#include "TMath.h"
+
+using namespace o2::trd;
+
+TrackletTransformer::TrackletTransformer()
+{
+  mgeo = o2::trd::Geometry::instance();
+  mgeo->createPadPlaneArray();
+}
+
+std::vector<double> TrackletTransformer::transformToSpacePoint(int hcid, int padrow, int column, int position)
```

Comment:
Okay, I split the transformation into three parts for each coordinate axis. I noticed that this involved some repetition of the padPlane calls. To avoid this, I created a new method in `TrackletTransformer` called `loadPadPlane` that initializes some stuff. I realize that this means that one can never call any of the transformations without also calling the `loadPadPlane` method so that is an extra step, but I think maybe it is cleaner this way? Let me know what you think. Will commit soon

Reply:
I am not sure how you implemented loadPadPlane, but couldn't you simply determine the chamber ID and pad plane in the tracklet loop in the `TRDTrackletTransformerSpec.cxx` and then pass a pointer of the pad plane to the respective functions `getY()` etc?

Reply:
I could do that, but I wanted to keep the `mGeo` instance out of `TRDTrackletTransformerSpec.cxx`